manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,386 @@
+         }
+         throw new IllegalArgumentException("Unsupported KeyType");
+      }
+
+      @Override
+      public String plainSeparator() {
+         return lazyRule.getKey().getKeySeparator();
+      }
+
+      @Override
+      public String selectStatement() {
+         return lazyRule.getQuery();
+      }
+      @Override
+      public String toString() {
+         return "LazyCacheRuleSpecAdapter{" +
+                 "keyType=" + keyType() +
+                 ",plainSeparator=" + plainSeparator() +
+                 ",selectStatement=" + selectStatement() +
+                 '}';
+      }
+   }
+
+   /** Adapt an EagerCacheRuleSpec to be and EagerRule */
+   static class EagerCacheRuleSpecAdapter implements EagerRule {
+      final EagerCacheRuleSpec eagerRule;
+      private String selectStmt;
+      private Connector connector;
+
+      public EagerCacheRuleSpecAdapter(EagerCacheRuleSpec eagerRule) {
+         this.eagerRule = eagerRule;
+      }
+
+      @Override
+      public Connector connector() {
+         if (connector == null) {
+            String[] schemaTable = eagerRule.getTableName().split("\\.");
+            if (schemaTable.length > 2) {
+               throw new IllegalArgumentException("Unsupported schema format (must be table or schema.table)");
+            }
+            connector = new Connector() {
+               @Override
+               public String schema() {
+                  return (schemaTable.length == 1) ? "" : schemaTable[0];
+               }
+
+               @Override
+               public String table() {
+                  return (schemaTable.length == 1) ? schemaTable[0] : schemaTable[1];
+               }
+            };
+         }
+         return connector;
+      }
+
+      @Override
+      public KeyType keyType() {
+         return switch (eagerRule.getKey().getFormat()) {
+            case TEXT -> KeyType.TEXT;
+            case JSON -> KeyType.JSON;
+            case UNRECOGNIZED -> throw new IllegalArgumentException("Unsupported KeyType");
+         };
+      }
+
+      @Override
+      public String plainSeparator() {
+         return eagerRule.getKey().getKeySeparator();
+      }
+
+      @Override
+      public String selectStatement() {
+         if (selectStmt == null) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("SELECT ")
+                  .append(
+                        eagerRule.getValue().getValueColumnsCount() > 0
+                              ? String.join(", ", eagerRule.getValue().getValueColumnsList())
+                              : "*")
+                  .append(" FROM ")
+                  .append(eagerRule.getTableName());
+            if (eagerRule.getKey().getKeyColumnsCount() > 0) {
+               sb.append(" WHERE ").append(
+                     eagerRule.getKey()
+                           .getKeyColumnsList()
+                           .stream()
+                           .map(s -> s + " = ?")
```

Comment:
I found another bug. This will only work with a DB that uses `?` as a placeholder, e.g. MySQL. We need DB vendor specific select statements.

Let's not delay this PR any longer though, I have created https://github.com/gingersnap-project/cache-manager/issues/56 so this can be addressed in a subsequent PR.